docs/EN_US/ECLStandardLibraryReference/SLR-Mods/Crypto.xml

```diff
@@ -0,0 +1,1323 @@
+
+        <tbody>
+          <row>
+            <entry>Return:<emphasis></emphasis></entry>
+
+            <entry>SET OF STRINGs containing all supported Public Key
+            Algorithms</entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable>
+
+    <para>The <emphasis role="bold">SupportedPublicKeyAlgorithms
+    </emphasis>function returns the set of supported Public Key
+    Algorithms</para>
+
+    <para>Example:</para>
+
+    <programlisting format="linespecific">IMPORT STD;
+STD.Crypto.SupportedPublicKeyAlgorithms(); //returns SET of STRINGs</programlisting>
+
+    <para></para>
+  </sect1>
+
+  <sect1 id="hashing">
+    <title>Hashing Module</title>
+
+    <para><emphasis>myHashModule</emphasis> :=<emphasis role="bold">
+    STD.Crypto.Hashing(</emphasis><emphasis>hashAlgorithm</emphasis><emphasis
+    role="bold">)<indexterm>
+        <primary>Hashing</primary>
+      </indexterm>;</emphasis></para>
+
+    <informaltable colsep="1" frame="all" rowsep="1">
+      <tgroup cols="2">
+        <colspec colwidth="120.50pt" />
+
+        <colspec />
+
+        <tbody>
+          <row>
+            <entry><emphasis>myHashModule</emphasis></entry>
+
+            <entry>The name of the Hashing module structure </entry>
+          </row>
+
+          <row>
+            <entry><emphasis>hashAlgorithm</emphasis></entry>
+
+            <entry>The hashing algorithm to use, as returned by <link
+            linkend="SupportedHashAlgorithms">SupportedHashAlgorithms()</link></entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable>
+
+    <para>A Hashing module is defined in ECL. Subsequent function definitions
+    use the module definitions specified in the Hashing module
+    definition.</para>
+
+    <para>Example:</para>
+
+    <programlisting format="linespecific">Import STD;
+
+//Hashing module definition
+myHashModuleSha512 := Std.Crypto.Hashing('sha512'); 
+myHashModuleSha256 := Std.Crypto.Hashing('sha256'); 
+
+DATA hash1 := myHashModuleSha512.Hash((DATA)'The quick brown fox jumps over the lazy dog');
+DATA hash2 := myHashModuleSha256.Hash((DATA)'The quick brown fox jumps over the lazy dog');
+
+OUTPUT(hash1);
+OUTPUT(hash2);</programlisting>
+  </sect1>
+
+  <sect1 id="hash">
+    <title>Hash</title>
+
+    <para><emphasis>myHashModule</emphasis><emphasis
+    role="bold">.Hash(</emphasis><emphasis>inputData</emphasis><emphasis
+    role="bold">)<indexterm>
+        <primary>Hash</primary>
+      </indexterm>;</emphasis></para>
+
+    <informaltable colsep="1" frame="all" rowsep="1">
+      <tgroup cols="2">
+        <colspec colwidth="120.50pt" />
+
+        <colspec />
+
+        <tbody>
+          <row>
+            <entry><emphasis>myHashModule</emphasis></entry>
+
+            <entry>The name of the Hashing module structure</entry>
+          </row>
+
+          <row>
+            <entry><emphasis>inputData</emphasis></entry>
+
+            <entry>The data to hash</entry>
```

Comment:
In our call with RT, we decided to specify the TYPE of the argument

Reply:
@RussWhitehead Made requested edits, back to you